analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java

```diff
@@ -644,4 +651,116 @@ private void writeRepositorySettingsFile(Map<ArtifactRepository, Path> repositor
+                    .entrySet();
+
+            for (Map.Entry<Project, Map<RelaxedProjectVersionRef, ProjectVersionRef>> entry : entrySet) {
+                final Project name = entry.getKey();
+                final Map<RelaxedProjectVersionRef, ProjectVersionRef> allModuleDependencies = entry.getValue();
+                final ManipulationModel correspondingModule = alignmentModel.findCorrespondingChild(name);
+                final String group = correspondingModule.getGroup().isEmpty() ? alignmentModel.getGroup()
+                        : correspondingModule.getGroup();
+                final String artifact = correspondingModule.getName();
+                final String ga = group + ":" + artifact;
+                final String v = correspondingModule.getOriginalVersion() == null ? originalVersion
+                        : correspondingModule.getOriginalVersion();
+                final String originalModuleGav = ga + ":" + v;
+                final String newModuleVersion = correspondingModule.getVersion();
+                final String newModuleGav = ga + ":" + newModuleVersion;
+                append(builder, "------------------- project {}", ga);
+
+                if (!v.equals(newModuleVersion)) {
+                    append(builder, "\tProject version : {} --> {}", v, newModuleVersion);
+                }
+
+                builder.append(System.lineSeparator());
+
+                final ModulesItem module = new ModulesItem();
+                module.getGav().setOriginalGAV(originalModuleGav);
+                module.getGav().setPVR(SimpleProjectVersionRef.parse(newModuleGav));
+                modules.add(module);
+                final Map<String, ProjectVersionRef> dependencies = module.getDependencies();
+                final boolean reportNonAligned = configuration.reportNonAligned();
+                allModuleDependencies.forEach((d, p) -> {
+                    final ProjectVersionRef newDependencyVersion = correspondingModule.getAlignedDependencies()
+                            .get(d.toString());
+                    logger.debug("In module {} with GAV {} found a replacement version of {}",
+                            correspondingModule.getProjectPathName(), p, newDependencyVersion);
+
+                    if (newDependencyVersion == null) {
+                        if (reportNonAligned) {
+                            nonAligned.add(d);
+                        }
+                    } else {
+                        append(builder, "\tDependencies : {} --> {}", p, newDependencyVersion);
+                        dependencies.put(d.toString(), newDependencyVersion);
+                    }
+                });
+            }
+
+            if (!nonAligned.isEmpty()) {
+                for (ProjectVersionRef na : nonAligned) {
+                    logger.debug("Non-Aligned Dependencies : {}", na);
+                    append(builder, "\tNon-Aligned Dependencies : {}", na);
+                }
+            }
```

Comment:
Question - is nonAligned reported on per sub-module/project basis or all together at the end? If the latter, could we add some spacing ?

Reply:
Good point---I think it needs to move up a line to 741 inside the `forEach` loop. It may be outside the loop now. Is there a multi-module project similar to simple-project that's ready to be used to test multi-module?

Reply:
Also, does it make sense to refactor this method to make it smaller?

Reply:
Re "moving" - agreed.
Re "refactoring" - not sure how you can make it smaller without adding extra methods. For that matter, the block after ` if (cache.removeProject(project)) {` could be moved to its own method perhaps. Probably not essential to do at the moment IMHO.

Re "multi-module project" - many of the functional tests are multimodule - but you probably want one with some dependencies to try - you could look at the spring ones?

Reply:
> Re "refactoring" - not sure how you can make it smaller without adding extra methods

I thought "refactoring" meant (possibly) adding/removing methods. Regardless, what I meant was moving some of the code to smaller methods instead of leaving a single large method.

Reply:
> Question - is nonAligned reported on per sub-module/project basis or all together at the end? If the latter, could we add some spacing ?

OK, I fixed this to move the `nonAligned` inside the loop. It should be reported together with the sub-modules dependencies from what I see in PME. In PME, there is no line between the aligned and non-aligned dependencies, but there is a line after all of them are printed. Sometimes, there are two spacer lines, but it's hard to tell when this in intentional.

There is already a single spacer line after the dependencies. But, right now I think it might be possible that if nothing gets printed it still prints a spacer line.

Reply:
Don't worry about the spacing - that was more if it _wasn't_ running inside the loop.

Reply:
Ok the non-aligned dependencies from my manual test do now appear next to submodules.

I think (for future!) we have to do more work on isolating non-aligned down to direct dependencies etc.